### PR TITLE
Adapters & Client refactor

### PR DESCRIPTION
**Pull request by Kogaroshi**
Creation of backend adapters
New Adapter : legacy_storage that contains the existing Client logic
Client now interfaces adapters

### CODE REVIEW

File: lib/sidekiq_flow/client.rb [hunk 1]
```diff
@@ -34,53 +34,31 @@ def clear_task(workflow_id, task_class)
...
 
       def find_task(workflow_id, task_class)
         find_workflow(workflow_id).find_task(task_class)
       end
 
       def find_workflow_keys(pattern=workflow_key_pattern)
-        connection_pool.with do |redis|
-          redis.scan_each(match: pattern, count: SCAN_COUNT).to_a
-        end
+        adapters.last.find_workflow_keys(pattern=workflow_key_pattern)
```
Comment (franee) [inline lib/sidekiq_flow/client.rb:61]: pass only pattern here.

**Review by franee — CHANGES_REQUESTED** [review]
Please also update the gem version.

File: lib/sidekiq_flow/client.rb (hunk 1 above)
Comment (VIGER-Valentin) [reply lib/sidekiq_flow/client.rb]: Done

File: lib/sidekiq_flow/adapters/legacy_storage.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            
+          def already_succeeded?(workflow_id, workflow_key)
+            workflow_key.match(/#{configuration.namespace}\.#{workflow_id}_\d{10}_\d{10}/)
+          end
+    
+        end
+      end
+    end
+  end
+
```
Comment (franee) [inline lib/sidekiq_flow/adapters/legacy_storage.rb:126]: nit, pls configure your editor to remove extra blank lines, or just run `rubocop --fix <path to your file>` before committing.

**Review by franee — APPROVED** [review]

**Pull request closed without merging**